sources/core/Stride.Core.IO/VirtualFileStream.cs

```diff
@@ -203,6 +203,17 @@ public override void Write(byte[] buffer, int offset, int count)
             InternalStream.Write(buffer, offset, count);
         }
 
+        /// <inheritdoc/>
+        public override void Write(ReadOnlySpan<byte> buffer)
+        {
+            if (endPosition != -1 && buffer.Length > endPosition - InternalStream.Position)
+            {
+                throw new NotSupportedException("Can't write beyond end of stream.");
```

Comment:
This should be an `InvalidOperationException`.
> The InvalidOperationException is thrown when a method call is invalid for the object's current state.

> The NotSupportedException is thrown when an invoked method is not supported, or when there is an attempt to read, seek, or write to a stream that does not support the invoked functionality.

Reply:
It is what `void Write(byte[] buffer, int offset, int count)` does. You want me to change them in both?

Reply:
Yes, please do. If I encountered this exception I'd be confused.

Reply:
Then I make it `IOException`. `System.IO.Stream` doesn't throw `InvalidOperationException` except for:

`Stream.WriteAsync`/`Stream.WriteAsync`:
* [InvalidOperationException](https://docs.microsoft.com/en-us/dotnet/api/system.invalidoperationexception?view=net-6.0)
The stream is currently in use by a previous read/write operation.

Making that consistent would affect `VirtualFileStream`, `LZ4Stream` and the obsolete `NativeMemoryStream`.